### PR TITLE
Tweak: I/E - break down import process into chunks [HTS-36]

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by ranshoshani34 — CHANGES_REQUESTED** [review]
looks really good! great job

File: app/modules/import-export/assets/js/hooks/use-kit.js [hunk 1]
```diff
@@ -51,7 +52,52 @@ export default function useKit() {
...
+
+				ajaxConfig.data.data = JSON.stringify( ajaxConfig.data.data );
+
+				// The last runner should run using the setAjax method, so it will trigger the useEffect and update the kitState.
+				const isLastIteration = iteration === runners.length - 1;
+				if ( ! isLastIteration ) {
+					await runRequest( ajaxConfig );
+				} else {
+					setAjax( ajaxConfig );
+				}
```
[the remark this answers is not in the document]
Comment (ranshoshani34) [reply app/modules/import-export/assets/js/hooks/use-kit.js:77]: After our last discussion this is the best solution

**Review by ranshoshani34 — DISMISSED** [review]

File: app/modules/import-export/processes/import.php [hunk 2]
```diff
@@ -136,33 +136,73 @@ class Import {
...
+			$this->settings_selected_plugins = $settings['plugins'] ?? null;
+
+			$this->manifest = $this->read_manifest_json();
+			$this->site_settings = $this->read_site_settings_json();
+
+			$this->set_default_settings();
 		}
+	}
+
+	private function set_import_object( $old_instance ) {
```
Comment (nivnoimanelementor) [inline app/modules/import-export/processes/import.php:172]: please change the name of the $old_instance to $instance -
thi function don't need to know that this is new or old instance.

**Review by nivnoimanelementor — CHANGES_REQUESTED** [review]
Good job, Pleas check the request changes.

Also, please update(add) all the method here with docBlock

File: app/modules/import-export/processes/import.php [hunk 3]
```diff
@@ -136,33 +136,73 @@ class Import {
 	 *      (e.g: include, selected_plugins, selected_cpt, selected_override_conditions, etc.)
 	 * @throws \Exception
 	 */
-	public function __construct( string $path, array $settings = [] ) {
-		if ( is_file( $path ) ) {
-			$this->extracted_directory_path = $this->extract_zip( $path );
+	public function __construct( string $path, array $settings = [], $old_instance = null ) {
```
[the remark this answers is not in the document]
Comment (ManorHazaz) [reply app/modules/import-export/processes/import.php:139]: Done

File: app/modules/import-export/processes/import.php (hunk 2 above)
Comment (ManorHazaz) [reply app/modules/import-export/processes/import.php:172]: Done

**Review by nivnoimanelementor — DISMISSED** [review]
Great Job

**Pull request closed without merging**